src/shogun/base/SGObject.h

```diff
@@ -1022,5 +1010,158 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+	template <class T>
+	class ObservedValueTemplated;
+
+	/**
+	 * Observed value which is emitted by algorithms.
+	 */
+	class ObservedValue : public CSGObject
+	{
+	public:
+		/**
+		 * Constructor
+		 * @param step step
+		 * @param name name of the observed value
+		 */
+		ObservedValue(int64_t step, std::string name);
+
+		/**
+		 * Destructor
+		 */
+		~ObservedValue(){};
+
+#ifndef SWIG
+		/**
+		* Return a any version of the stored type.
+		* @return the any value.
+		*/
+		virtual Any& get_any()
+		{
+			return m_any_value;
+		}
+#endif
+
+		/** @return object name */
+		virtual const char* get_name() const
+		{
+			return "ObservedValue";
+		}
+
+	protected:
+		/** ObservedValue step (used by Tensorboard to print graphs) */
+		int64_t m_step;
+		/** Parameter's name */
+		std::string m_name;
+		/** Untyped value */
+		Any m_any_value;
+	};
+
+	/**
+	 * Templated specialisation of ObservedValue that stores the actual data.
+	 * @tparam T the type of the observed value
+	 */
+	template <class T>
+	class ObservedValueTemplated : public ObservedValue
+	{
+
+	public:
+		/**
+		 * Constructor
+		 * @param step step
+		 * @param name the observed value's name
+		 * @param value the observed value
+		 */
+		ObservedValueTemplated(int64_t step, std::string name, std::string description, T value)
+				: ObservedValue(step, name), m_observed_value(value)
+		{
+			this->watch_param(
+					name, &m_observed_value,
```

Comment:
The name of the ObservedValue now matches the tag's one. The new behaviour is:
```c++
ObservedValue observation;

// Instead of using get<T>("value") we do
observation->get<SGVector<float64_t>>("w")
```

Reply:
that is nice!